llvm/include/llvm/IR/Constants.h

```diff
@@ -136,7 +144,11 @@ class ConstantInt final : public ConstantData {
   inline const APInt &getValue() const { return Val; }
 
   /// getBitWidth - Return the bitwidth of this constant.
-  unsigned getBitWidth() const { return Val.getBitWidth(); }
+  unsigned getBitWidth() const {
+    assert(getType()->isIntegerTy() &&
+           "Returning the bitwidth of a vector constant is not support!");
```

Comment:
Why? I think this API should work with vectors as well (with the current implementation, i.e. returning the bitwidth of the  scalar value). You can just adjust the comment to clarify.

Reply:
IMHO this is not necessary. If the name were `getSizeInBits()` I would agree, but the term "bit width" implies that we're talking about scalar. We don't use the term "bit width" to refer to full size of a vector.

Reply:
Fair enough.